Automattic/jetpack
# Jetpack: improve the performance of the disconnect dialog

The plugins page load is slow with Jetpack activated - it has been suggested that a lot of time is spent on the PHP side initialising state for the React disconnect modal.

#### Changes proposed in this Pull Request:

1. This PR stops using the main Jetpack app's PHP initial state helper, and instead fetches the data it needs async from the React side.
2. Reduced the size of the redux store and state required by the app - saves 100+Kb raw, 10Kb Gzipped

This is aimed all at improving the performance and load time of the wp-admin plugins page - no longer making all the requests that were being made for the full blown React initial state.

The loading of data is not necessary to allow the user to disconnect.

Slow user data fetching will prevent tracking events on the dialog clicks, but that usually won't happen and I'm not sure we should make the user wait before they can disconnect in those rare cases where fetch requests are slow.

Also, slow requests for connected plugins, or site benefits result in the following fallback display:
<img width="718" alt="fall_back" src="https://user-images.githubusercontent.com/8742105/154708831-08de4c10-be67-4716-9688-523c4f3ff2fa.png">

As a result, I don't think it's worth trying to show any busy text or spinners. The data loads in the background before opening the modal, and the data isn't critical to allow the user to disconnect, and the fallback display looks good anyway.

#### Jetpack product discussion

p1HpG7-ere-p2#comment-52144

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

Performance testing will be done separately to ensure this PR is worthwhile - but basic navigation/refreshing of the admin Plugins page should certainly not take any longer to load than usual, even if there is no immediately obvious speed increase.

We should perform some regression testing to ensure the disconnect modal works as it did previously from the Plugins page.

- the modal from Jetpack dashboard Managing site connection **should remain unchanged.**
- note that the disconnect modal shown on the plugins page disconnects immediately and refreshes the page without showing the reason for disconnect, and thank you steps.
- the plugins page disconnect modal may show placeholders while it waits for data to be fetched (emulate a slow network if necessary)
- otherwise, the plugins page disconnect modal should behave as before
